### PR TITLE
Further updates to `tbl_hierarchical()`

### PR DESCRIPTION
**Pull request by ddsjoberg**
**What changes are proposed in this pull request?**
<< Insert text here that can be directly copied into NEWS.md by your reviewer. >>

**If there is an GitHub issue associated with this pull request, please provide link.**

--------------------------------------------------------------------------------

Reviewer Checklist (if item does not apply, mark is as complete)

- [ ] PR branch has pulled the most recent updates from main branch.
- [ ] If a bug was fixed, a unit test was added.
- [ ] Run `pkgdown::build_site()`. Check the R console for errors, and review the rendered website.
- [ ] Code coverage is suitable for any new functions/features: `devtools::test_coverage()`
- [ ] `usethis::use_spell_check()` runs with no spelling errors in documentation
- [ ] **All** GitHub Action workflows pass with a :white_check_mark:

When the branch is ready to be merged into master:
- [ ] Update `NEWS.md` with the changes from this pull request under the heading "`# gtsummary (development version)`". If there is an issue associated with the pull request, reference it in parentheses at the end update (see `NEWS.md` for examples).
- [ ] Increment the version number using `usethis::use_version(which = "dev")` 
- [ ] Run `usethis::use_spell_check()` again
- [ ] Approve Pull Request
- [ ] Merge the PR. Please use "Squash and merge".

### CODE REVIEW

File: R/brdg_hierarchical.R [hunk 1]
```diff
@@ -61,11 +61,10 @@ brdg_hierarchical <- function(cards,
     cards <- cards |>
       mutate(
         variable_level = ifelse(
-          .data$variable == "..ard_hierarchical_overall..", label[["overall"]], .data$variable_level
```
Comment (ddsjoberg) [inline R/brdg_hierarchical.R:64]: FYI I think it'll be easier in the long run to retain the name from the ARD output, and the name overall may be too common and cause some name conflicts in the future

File: R/tbl_hierarchical.R [hunk 2]
```diff
@@ -99,14 +100,13 @@
 #'   by = TRTA,
 #'   overall_row = TRUE
 #' )
-#'
 tbl_hierarchical <- function(data,
                              variables,
+                             id,
+                             denominator,
                              by = NULL,
-                             id = NULL,
```
Comment (ddsjoberg) [inline R/tbl_hierarchical.R:106]: FYI I removed the default values for arguments that are required for the users to pass.

**Review by edelarua — CHANGES_REQUESTED** [review]
A few small suggestions but otherwise all good!! :)

File: R/brdg_hierarchical.R [hunk 3]
```diff
@@ -61,11 +61,10 @@ brdg_hierarchical <- function(cards,
...
+          .data$variable == "..ard_hierarchical_overall..", label[["..ard_hierarchical_overall.."]], .data$variable_level
         ),
-        variable = ifelse(.data$variable == "..ard_hierarchical_overall..", "overall", .data$variable)
+        variable = ifelse(.data$variable == "..ard_hierarchical_overall..", "..ard_hierarchical_overall..", .data$variable)
```
[the remark this answers is not in the document]
Comment (ddsjoberg) [reply R/brdg_hierarchical.R:66]: haha, thanks!

**Pull request merged**